src/Stratis.Bitcoin.Features.PoA/Voting/JoinFederationRequestService.cs

```diff
@@ -92,16 +72,76 @@ public async Task<PubKey> JoinFederationAsync(JoinFederationRequestModel request
                 throw new Exception($"The call to sign the join federation request failed: '{err.Message}'.");
             }
 
+            if (!VerifyCollateralSignature(joinRequest))
```

Comment:
You check this inside `BroadcastSignedJoinRequestAsync` too - as that method gets called in two places it makes sense to keep the `if` inside it only